students/johnpharmd/lesson04/trigrams.py

```diff
@@ -28,30 +28,86 @@
 
 # Data
 sample_st = 'I wish I may I wish I might'
+read_data = []
 s1 = ''
 s2 = ''
+st_key = ''
 s3 = ''
-count = 0
-st_set = set()
 st_dict = {}
+count = 0
 pron_set = set(['I', 'you', 'she', 'he', 'it', 'we', 'they'])
-
+n = 0
+new_st_lst= []
+testfile_dict = {1: 'sherlock_small.txt', 2: 'sherlock.txt'}
 
 # Processing
-st_split_lst = sample_st.split(' ')
-print('st_split_lst: ', st_split_lst, '\n')
-for i, word in enumerate(st_split_lst):
-    if i + 2 < len(st_split_lst):
-        s1 = word
-        s2 = st_split_lst[i + 1]
-        s3 = st_split_lst[i + 2]
-        st_set.update([s1, s2])
-        st_dict[count] = {frozenset(st_set): s3}
-        # print('st_set:', st_set)
-        count += 1
-        s1 = ''
-        s2 = ''
-        s3 = ''
-        st_set = set()
-for k, v in st_dict.items():
-    print(k, v, '\n')
+def clean_data(choice=1):
+    # use 2 as param, if want to test with 'sherlock.txt'
+    with open(testfile_dict[choice], 'r') as f:
+    # header_size = 0
+    # with open('sherlock.txt') as f:
+        read_data = f.read()
+        # read_data = f.seek('To Sherlock Holmes she')
+    read_data = ''.join(read_data.split('\r'))
+    read_data = ''.join(read_data.split('"'))
+    read_data = ' '.join(read_data.split('\n'))
+    read_data = ' '.join(read_data.split('--'))
+    read_data = ' '.join(read_data.split('-'))
+    read_data = ' '.join(read_data.split(', '))
+    read_data = ' '.join(read_data.split('. '))
+    read_data = ''.join(read_data.split('('))
+    read_data = ''.join(read_data.split(')'))
+    read_data = read_data.strip('.')
+    read_data = read_data.strip('  ')
+    read_data = read_data.split(' ')
```

Comment:
you could rewrite your join/split code into a loop using `replace` with replace char map
```
replace_map = {"": ("\r", '"'), " ": ("(", ")")}
for new_char, chars in replace_map.items():
    for c in chars:
        read_data = read_data.replace(c, new_char)
```